### PR TITLE
Get ETABS File Path

### LINKED ISSUE #413: ETABS filname/filepath
#### Description:

Can the filename of the attached instance of ETABS be exposed through BHoM? Perhaps as a Query on the adapter.
The following API method would do the trick.

ETABSv1.cSapModel.GetModelFilename()

### PR DESCRIPTION
**Pull request by GCRA101**
### Issues addressed by this PR

 Closes #413 

GH Script - Canvas View

ETABS Toolkit now allows to retrieve the full filepath of the ETABS Model as a public attribute of the ETABSAdapter class.

 ### Test files
Grasshopper File

ETABS File

 ### Changelog
- Added public and read-only attribute FilePath to the ETABSAdapter Class
- Added assignment of the FilePath attribute in the Constructor of the ETABSAdapter Class

### CODE REVIEW

**Comment by Chrisshort92** [comment]
Testing against the following test scripts as link in PR above is not live; [https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/ETABS_Toolkit/%23478-GetEtabsFilePath?csf=1&web=1&e=Z4BiJY](url)

**Review by Chrisshort92 — APPROVED** [review]
Functionality tested against test files in [https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/ETABS_Toolkit/%23478-GetEtabsFilePath?csf=1&web=1&e=Z4BiJY](url). Working as expected, approved for merge

**Comment by Chrisshort92** [comment]
@BHoMBot check required

**Comment by peterjamesnugent** [comment]
@BHoMBot check copyright-compliance

**Comment by GCRA101** [comment]
@BHoMBot check ready-to-merge

File: ETABS_oM/ETABS_oM.csproj [hunk 1]
```diff
@@ -49,8 +49,8 @@
     </Reference>
     <Reference Include="Quantities_oM">
       <HintPath>$(ProgramData)\BHoM\Assemblies\Quantities_oM.dll</HintPath>
-	  <Private>False</Private>
-	  <SpecificVersion>False</SpecificVersion>
+      <Private>False</Private>
+      <SpecificVersion>False</SpecificVersion>
```
Comment (peterjamesnugent) [inline ETABS_oM/ETABS_oM.csproj:53]: Can we remove this change - it's not part of the PR.

File: Etabs_Adapter/ETABSAdapter.cs [hunk 2]
```diff
@@ -138,6 +140,8 @@ public ETABSAdapter(string filePath = "", EtabsSettings etabsSetting = null, boo
                         m_model.File.NewBlank();
                 }
 
+                FilePath = m_model.GetModelFilename();
```
Comment (peterjamesnugent) [inline Etabs_Adapter/ETABSAdapter.cs:143]:
```suggestion
                FilePath = m_model.GetModelFilename();
```

Has this been tested after using the `Execute` action to create a new file - it's common that they won't have a filename until you SaveAs - what happens then?

**Review by peterjamesnugent — CHANGES_REQUESTED** [review]
Just one change needed, and maybe some further testing.

File: ETABS_oM/ETABS_oM.csproj (hunk 1 above)
Comment (GCRA101) [reply ETABS_oM/ETABS_oM.csproj]: @peterjamesnugent,
not sure what's the change the Git has found in this. 
Opened the .csproj file in notepad++ and indentation is all fine and matching with all other corresponding lines below (see screenshot)

image

File: Etabs_Adapter/ETABSAdapter.cs (hunk 2 above)
Comment (GCRA101) [reply Etabs_Adapter/ETABSAdapter.cs:143]: Hi @peterjamesnugent, thanks a lot for this comment.
I've tested the code with Execute and added a few lines of code that allow to update the FilePath as soon as the new un-named model gets saved.
When the model is unsaved, the FilePath returns a "(Untitled)" string to signify that the model has no path currently.
All saved in latest commit ;)
you can find updated test GH script at same location (see below)
https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/ETABS_Toolkit/%23478-GetEtabsFilePath/Test%20Script.gh?csf=1&web=1&e=dEf085

**Review by peterjamesnugent — APPROVED** [review]
Changes since @Chrisshort92 review do not affect functionality, so approving.

**Comment by peterjamesnugent** [comment]
@BHoMBot check required
@BHoMBot check copyright-compliance

**Comment by GCRA101** [comment]
@BHoMBot check ready-to-merge
